packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,566 @@
+import BIP32Factory, { BIP32Interface } from 'bip32'
+import { derivePath } from 'ed25519-hd-key'
+import * as ecc from 'tiny-secp256k1'
+// eslint-disable-next-line ordered-imports/ordered-imports
+import {
+  AssetXRD,
+  MAINNET_GATEWAY_URL,
+  RadixChain,
+  STOKENET_GATEWAY_URL,
+  TransferTransactionManifest,
+  XRD_DECIMAL,
+  xrdRootDerivationPaths,
+} from './const'
+
+/**
+ * Custom Radix client
+ */
+
+export default class Client extends BaseXChainClient {
+  gatewayApiClient: GatewayApiClient
+  curve: Curve
+  constructor(params: XChainClientParams, curve: Curve) {
+    super(RadixChain, {
+      network: params.network,
+      phrase: params.phrase,
+      rootDerivationPaths: xrdRootDerivationPaths,
+      feeBounds: params.feeBounds,
+    })
+    this.curve = curve
+    this.gatewayApiClient = GatewayApiClient.initialize({
+      networkId: this.getRadixNetwork(),
+      applicationName: 'xchainjs',
+    })
+  }
+
+  /**
+   * Get an estimated fee for a test transaction that involves sending
+   * XRD from one account to another
+   *
+   * @returns {Fee} An estimated fee
+   */
+  async getFees(): Promise<Fees> {
+    try {
+      const gatewayStatusResponse: GatewayStatusResponse = await this.gatewayApiClient.status.getCurrent()
+      const transactionPreviewRequest: TransactionPreviewRequest = {
+        manifest: TransferTransactionManifest,
+        start_epoch_inclusive: gatewayStatusResponse.ledger_state.epoch,
+        end_epoch_exclusive: gatewayStatusResponse.ledger_state.epoch + 10,
+        tip_percentage: 10,
+        nonce: Math.floor(Math.random() * 1000000),
+        signer_public_keys: [
+          {
+            key_type: 'EddsaEd25519',
+            key_hex: this.getRadixPrivateKey().publicKey().hex(),
+          },
+        ],
+        flags: {
+          use_free_credit: true,
+          assume_all_signature_proofs: false,
+          skip_epoch_check: true,
+        },
+      }
+      const transactionPreviewResponse: TransactionPreviewResponse =
+        await this.gatewayApiClient.transaction.innerClient.transactionPreview({
+          transactionPreviewRequest: transactionPreviewRequest,
+        })
+      const receipt = transactionPreviewResponse.receipt as {
+        fee_summary: {
+          execution_cost_units_consumed: number
+          finalization_cost_units_consumed: number
+          xrd_total_execution_cost: string
+          xrd_total_finalization_cost: string
+          xrd_total_royalty_cost: string
+          xrd_total_storage_cost: string
+          xrd_total_tipping_cost: string
+        }
+      }
+      const totalFees =
+        parseFloat(receipt.fee_summary.xrd_total_execution_cost) +
+        parseFloat(receipt.fee_summary.xrd_total_finalization_cost) +
+        parseFloat(receipt.fee_summary.xrd_total_royalty_cost) +
+        parseFloat(receipt.fee_summary.xrd_total_storage_cost) +
+        parseFloat(receipt.fee_summary.xrd_total_tipping_cost)
+      const estimatedFees: Fees = {
```

Comment:
Few things:

1. Can we use the `singleFee` function here like other clients.
2. We have 18 decimal places, why are we not passing that to `assetAmount`? 
3. Are we sure about the call to `assetToBase` after `assetAmount`?

Reply:
1. yes
2. passing it now
3. returning this now
```const singleTxFee = baseAmount(totalFees, 18)
        return singleFee(FeeType.FlatFee, singleTxFee)
        ```